components/site/src/lib.rs

```diff
@@ -599,14 +599,27 @@ impl Site {
         for page in self.pages.values() {
             for alias in &page.meta.aliases {
                 let mut output_path = self.output_path.to_path_buf();
-                for component in alias.split('/') {
+                let mut split = alias.split('/').collect::<Vec<_>>();
+
+                // If the alias ends with an html file name, use that instead of mapping
+                // as a path containing an `index.html`
+                let page_name = match split.pop() {
+                    Some(part) if part.ends_with(".html") => part,
+                    Some(part) => {
+                        split.push(part);
+                        "index.html"
+                    }
+                    None => "index.html"
+                };
+
+                for component in split {
```

Comment:
Also @Keats, would it make sense to use [`create_dir_all`](https://doc.rust-lang.org/std/fs/fn.create_dir_all.html) here, rather than iterating over the path components?

Reply:
Go for it